docs/components/FSXABaseComponent.md

```diff
@@ -0,0 +1,128 @@
+[Readme](../../README.md) » [BaseComponents](../README.md) » FSXABaseComponent
+
+# FSXABaseComponent
+
+1. [Introduction](#introduction)
+2. [Usage](#usage)
+3. [Properties](#properties)
+4. [Methods](#methods)
+
+## Introduction
+
+The FSXABaseComponent is there to have a basic component to be able to implement further components.
+Thereby different attributes and methods are given, which should simplify the development significantly. These are described in this section.
+
+## Usage
+
+To use this base component a new class has to be created which extends the `FSXABaseComponent`.
+
+```tsx
+@Component
+class Component extends FSXABaseComponent {}
+```
+
+<hr>
+If you want to have type support for your props you can provide an interface and pass it as an generic like here the `Props` interface.
+
+```tsx
+@Component
+class Component extends FSXABaseComponent<Props> {}
+```
+
+<hr>
+If you want additionally type support for your events, you can pass a second interface.
+
+```tsx
+@Component
+class LayoutComponent extends FSXABaseComponent<Props, EventsWithOn> {}
+```
+
+Note: If you only want type support for your events and not the props you have to pass an empty interface `{}` as the data generic.
+
+<hr>
+To have additionally type support for your slots, you can pass a third interface.
+
+```tsx
+@Component
+class LayoutComponent extends FSXABaseComponent<Props, EventsWithOn, Slots> {}
+```
+
+Note: If you only want type support for your slots and not the props or events you have to pass empty interfaces `{}` as the `Props` and `EventsWithOn` generic.
+
+## Properties
+
+### `currentPage` - string
+
+This property returns the NavigationItem that is matching the current path. If null is returned, no current route could be matched to the current path.
+
+### `fsxaApi` - Object
+
+This property returns a preconfigured and ready-to-use FSXA-API instance.
+
+### `globalSettings` - string
+
+This property returns the globally configured project properties. If no projectProperties are defined in your CaaS this will return `null`.
+
+### `isEditMode` - boolean
+
+This property returns true if the app is delivering preview data. It will return false if it is released data.
+
+### `locale` - string
+
+This property returns the current locale abbreviation the content is displayed in e.g. `en_GB`.
+
+### `navigationData` - Object
+
+This property returns the current navigation data.
+
+## Methods
+
+### `getStoredItem`
+
+```typescript
+getStoredItem<T>(key: string): T
+```
+
+This method returns the stored data, which is set with the given key.
+
```

Comment:
"set with the given key"? Maybe "identified (by)"? I'd also remove the comma from this sentence.

Reply:
I have changed it to `which was set`. I want to make it clear, that the key needs to be set before.